src/Component/src/Factory/Factory.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Factory;
+
+/**
+ * Creates resources based on theirs FQCN.
+ */
+final class Factory implements FactoryInterface
+{
+    /**
+     * @var string
+     * @psalm-var class-string
+     */
+    private $className;
+
+    /**
+     * @psalm-param class-string $className
```

Comment:
Outside of scope of current PR,

Those `psalm-param` and `psalm-var` looks like a legacy solution for times when phpstan didn't have class-string and it can be updated now. (I don't have strict knowledge about this, just a feeling)